src/infra/repos/postgresql/size-repository.ts

```diff
@@ -1,11 +1,27 @@
 import { prisma as db } from './prisma'
-import { type LoadSizeByNameRepository } from '@/data/protocols/db/size'
+import { type LoadDogSizesRepository, type LoadCatSizesRepository, type LoadSizeByNameRepository } from '@/data/protocols/db/size'
 
-export class SizeRepository implements LoadSizeByNameRepository {
+export class SizeRepository implements LoadSizeByNameRepository, LoadCatSizesRepository, LoadDogSizesRepository {
   async loadByName (name: LoadSizeByNameRepository.Params): Promise<LoadSizeByNameRepository.Result> {
     const size = await db.size.findUnique({ where: { name } })
     if (size) {
       return size
     }
   }
+
+  async loadCatSizes (): Promise<LoadCatSizesRepository.Result> {
+    const sizes = await db.size.findMany({ where: { specie: { name: 'Gato' } } })
+    if (sizes.length > 0) {
+      return sizes
+    }
+    return []
+  }
+
+  async loadDogSizes (): Promise<LoadDogSizesRepository.Result> {
+    const sizes = await db.size.findMany({ where: { specie: { name: 'Cachorro' } } })
+    if (sizes.length > 0) {
+      return sizes
+    }
+    return []
+  }
```

Comment:
I believe it can be returned directly, because if it's empty it will already return an empty array and if it has something it will return the data